WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigActivity.kt

```diff
@@ -1,17 +1,145 @@
+            Log.d(TAG, "addOnChangeListener(): $slider, $value, $fromUser")
+            viewModel.setMinuteHandArmLength(value)
+        }
+
+        lifecycleScope.launch(Dispatchers.Main.immediate) {
+            viewModel = WatchFaceConfigViewModel(
+                    activity = this@WatchFaceConfigActivity,
+                    editIntent = intent
+                )
+
+            viewModel.uiState.collect { uiState: WatchFaceConfigViewModel.EditWatchFaceUiState ->
+                when (uiState) {
+                    is WatchFaceConfigViewModel.EditWatchFaceUiState.Loading -> {
+                        Log.d(TAG, "StateFlow Loading: ${uiState.message}")
+                    }
+                    is WatchFaceConfigViewModel.EditWatchFaceUiState.Success -> {
+                        Log.d(TAG, "StateFlow Success.")
+                        updateWatchFacePreview(uiState.userStylesAndPreview)
+                    }
+                    is WatchFaceConfigViewModel.EditWatchFaceUiState.Error -> {
+                        Log.e(TAG, "Flow error: ${uiState.exception}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun updateWatchFacePreview(
+        userStylesAndPreview: WatchFaceConfigViewModel.UserStylesAndPreview
+    ) {
+        Log.d(TAG, "updateWatchFacePreview: $userStylesAndPreview")
+
+        val colorStyleId: String = userStylesAndPreview.colorStyleId
+        Log.d(TAG, "\tselected color style: $colorStyleId")
+
+        binding.ticksEnabledSwitch.isChecked = userStylesAndPreview.ticksEnabled
+        binding.minuteHandLengthSlider.value = userStylesAndPreview.minuteHandLength
+        binding.preview.watchFaceBackground.setImageBitmap(userStylesAndPreview.previewImage)
+
+        if (!initialDataLoaded) {
+            initialDataLoaded = true
+            enabledWidgets()
+        }
+    }
+
+    private fun enabledWidgets() {
+        binding.colorStylePickerButton.isEnabled = true
+        binding.ticksEnabledSwitch.isEnabled = true
+        binding.minuteHandLengthSlider.isEnabled = true
+    }
+
+    override fun onDestroy() {
+        Log.d(TAG, "onDestroy()")
+        // Makes sure the activity closes.
+        finish()
+        super.onDestroy()
+    }
+
+    fun onClickColorStylePickerButton(view: View) {
+        Log.d(TAG, "onClickColorStylePickerButton() $view")
+
+        // TODO (codingjeremy): Replace with a RecyclerView to choose color style (next CL)
+        // Selects a random color style from list.
+        val colorStyleIdAndResourceIdsList = enumValues<ColorStyleIdAndResourceIds>()
+        val numbers: IntRange = colorStyleIdAndResourceIdsList.indices
+
+        val random = numbers.random()
+        val newColorStyle = colorStyleIdAndResourceIdsList[random]
+        val newColorStyleId = newColorStyle.id
+
+        Log.d(TAG, "\tnewColorStyleId (random): $newColorStyleId")
+
+        val newColorStyleValue = ColorStyleIdAndResourceIds.toOption(
```

Comment:
Could we instead try and get the right Option out of the UserStyleSchema rather than constructing a new one?

The EditorSession within WatchFaceConfigViewModel has a userStyleSchema property.  The UserStyleSchema is defined like so:

public class UserStyleSchema(
    public val userStyleSettings: List<UserStyleSetting>
)

It should be possible to find the color settings list from userStyleSettings (the editor should know the ID) on init. And then we can just iterate over the options list in the colorUserStyleSetting with findOrNull looking for a matching ID.

The advantage of this is we'd only need one place in the watch face code where the options are constructed, reducing the effort needed if a new color option was added.

Reply:
Done